### ISSUE TITLE
[Postgres] Make insert operations in postgres implementation similar to cassandra impl's

### ISSUE THREAD

**Issue by hungphan227** [linagora/james-project#5210]
Currently almost all insert operations in Cassandra implementation act like upsert. Meanwhile, insert operations in Postgres implementation does not allow inserting duplicate key. Probably we should make insert operations in postgres implementation similar to cassandra impl's so that behavior of postgres impl and cassandra impl are the same.

Few concerns:
- In postgres, to write "upsert" command, "on conflict" need to be used. However, it could cause performance issue (need to investigate)
- Adding onConflict will make the the code a bit longer

**Comment by hungphan227** [comment]
I have tested the performance of "on conflict do nothing" and it is the same as normal insert. IMO we can use "on conflict do nothing" to make postgres impl's behavior similar to cassandra impl's

**Comment by Arsnael** [comment]
I disagree honestly. I feel like we have more of a concurrency problem in our tests on tmail. Ignoring a duplicate instead of reporting it as an error isn't the solution I think

**Comment by chibenwa** [comment]
Indepotence most of the time makes sense except in very specific places...

**Comment by Arsnael** [comment]
Regarding the states for email and mailbox changes, I don't think it does..?

But for some other tables it might make sense yes
